src/character.cpp

```diff
@@ -714,9 +713,62 @@ double Character::aim_speed_dex_modifier() const
     return get_dex() - 8;
 }
 
-double Character::aim_speed_encumbrance_modifier() const
+float Character::aim_speed_modifier() const
 {
-    return ( encumb( body_part_hand_l ) + encumb( body_part_hand_r ) ) / 10.0;
+    return manipulator_score();
+}
+
+float Character::melee_thrown_move_modifier_hands() const
+{
+    if( manipulator_score() == 0.0f ) {
+        return MAX_MOVECOST_MODIFIER;
+    } else {
+        return std::min( MAX_MOVECOST_MODIFIER, 1.0f / manipulator_score() );
+    }
+}
+
+float Character::melee_thrown_move_modifier_torso() const
+{
+    const bodypart *torso = get_part( body_part_torso );
+    // TODO: make this "balance" perhaps?
+    if( torso == nullptr || torso->encumb_adjusted_limb_value( 1.0f ) == 0.0f ) {
+        return MAX_MOVECOST_MODIFIER;
+    } else {
+        return std::min( MAX_MOVECOST_MODIFIER, 1.0f / torso->encumb_adjusted_limb_value( 1.0f ) );
+    }
+}
+
+float Character::melee_stamina_cost_modifier() const
+{
+    if( lifting_score( body_part_type::type::arm ) == 0.0f ) {
+        return MAX_MOVECOST_MODIFIER;
+    } else {
+        return std::min( MAX_MOVECOST_MODIFIER, 1.0f / lifting_score( body_part_type::type::arm ) );
+    }
+}
+
+float Character::reloading_move_modifier() const
+{
+    if( manipulator_score() == 0.0f ) {
+        return MAX_MOVECOST_MODIFIER;
+    } else {
+        return std::min( MAX_MOVECOST_MODIFIER, 1.0f / manipulator_score() );
+    }
+}
+
+float Character::thrown_dex_modifier() const
+{
+    return manipulator_score();
+}
+
+float Character::ranged_dispersion_modifier() const
+{
+    if( manipulator_score() == 0.0f ) {
```

Comment:
```suggestion
    if( manipulator_score() <= std::numeric_limits<float>::epsilon() ) {
```
Note: May need an added (system) include file.

Reply:
while technically true, wouldn't this slightly obscure what this boolean statement is actually doing?

Reply:
Well, one could use `( 22.8 / FLT_MAX )` (or whatever the C++ idiomatic way to put it is), since we should try to avoid not only division by 0, but a result of the division greater than the maximum float. (Yes, the operation will likely be done as doubles, but subtracting 22.8 from something over the maximum float is not likely to bring it below the maximum float.)